pkgs/by-name/ro/roslyn-ls/package.nix

```diff
@@ -1,9 +1,9 @@
 { lib, fetchFromGitHub, buildDotnetModule, dotnetCorePackages, stdenvNoCC, testers, roslyn-ls, jq }:
 let
   pname = "roslyn-ls";
-  dotnet-sdk = dotnetCorePackages.sdk_9_0;
+  dotnet-sdk = with dotnetCorePackages; combinePackages [ sdk_6_0 sdk_7_0 sdk_8_0 sdk_9_0 ];
```

Comment:
roslyn-ls uses the runtime packages from dotnet 6, 7, 8, and 9.  Including 7 here removes the dependency on its package.  Unfortunately 6 and 8 are still present using older versions (6.0.32 and 8.0.7).  

I _think_ this is because sdk 9 preview 7 has hardcoded references to 6.0.32, 7.0.20, and 8.0.7, but I haven't had a chance to go digging for that yet.

I'm not sure what the best solution to this is.

Reply:
This might be an option:

```
diff --git a/pkgs/by-name/ro/roslyn-ls/package.nix b/pkgs/by-name/ro/roslyn-ls/package.nix
index 3207eea29d09..5f5096f28f4d 100644
--- a/pkgs/by-name/ro/roslyn-ls/package.nix
+++ b/pkgs/by-name/ro/roslyn-ls/package.nix
@@ -1,9 +1,9 @@
 { lib, fetchFromGitHub, buildDotnetModule, dotnetCorePackages, stdenvNoCC, testers, roslyn-ls, jq }:
 let
   pname = "roslyn-ls";
-  dotnet-sdk = with dotnetCorePackages; combinePackages [ sdk_6_0 sdk_7_0 sdk_8_0 sdk_9_0 ];
+  dotnet-sdk = dotnetCorePackages.sdk_9_0;
   # need sdk on runtime as well
-  dotnet-runtime = dotnetCorePackages.sdk_9_0;
+  dotnet-runtime = dotnet-sdk;
 
   project = "Microsoft.CodeAnalysis.LanguageServer";
 in
@@ -45,6 +45,14 @@ buildDotnetModule rec {
     # see this comment: https://github.com/NixOS/nixpkgs/pull/318497#issuecomment-2256096471
     # we can remove below line after https://github.com/dotnet/roslyn/issues/73439 is fixed
     "-p:UsingToolMicrosoftNetCompilers=false"
+    # force everything to net9.0
+    "-p:NetRoslyn=net9.0"
+    "-p:NetRoslynSourceBuild=net9.0"
+    "-p:NetRoslynAll=net9.0"
+    "-p:NetRoslynBuildHostNetCoreVersion=net9.0"
+    "-p:NetRoslynNext=net9.0"
+    "-p:NetVSCode=net9.0"
+    "-p:NetVSShared=net9.0"
   ];
 
   # two problems solved here:
```

@konradmalik what would you think of that?  it allows all the reference dependencies to be removed.

Reply:
Another option is to look into forcing `KnownAppHost` to use the versions we actually have in nixpkgs.

Reply:
I'm fine with that if it works. Do you want to include it here, or should I create a PR? could try KnownAppHost there as well.

Reply:
We could also just leave it as @Smaug123 suggested. Not sure what it means for the deprecated dotnet 7 though.

Reply:
> Also an idea - we could try setting props such that [this branch will be used](https://github.com/dotnet/roslyn/blob/d176f9b5a7220cd95a6d5811ba1c49ac392a2fdc/eng/targets/TargetFrameworks.props#L53)

I did try this, but `DotNetBuildFromSource` seems to cause other problems.  I think it might be meant for building as part of the VMR?

I think I'll leave it as is for now.  If we ever improve `KnownAppHost` in the build infrastructure, the dependencies will be reduced on the next fetch-deps, but only if we used the combined sdks.